src/frames.rs

```diff
@@ -7,85 +7,55 @@ use std::os::raw::c_void;
 use std::path::PathBuf;
 
 use backtrace::Frame;
+use smallvec::SmallVec;
 use symbolic_demangle::demangle;
 
 use crate::{MAX_DEPTH, MAX_THREAD_NAME};
 
-#[derive(Debug, Clone)]
-pub struct UnresolvedFramesSlice<'a> {
-    pub frames: &'a [Frame],
-    pub thread_name: &'a [u8],
-    pub thread_id: u64,
-}
-
+#[derive(Clone)]
 pub struct UnresolvedFrames {
-    pub frames: [Frame; MAX_DEPTH],
-    pub depth: usize,
+    pub frames: SmallVec<[Frame;MAX_DEPTH]>,
     pub thread_name: [u8; MAX_THREAD_NAME],
     pub thread_name_length: usize,
     pub thread_id: u64,
 }
 
 impl Default for UnresolvedFrames {
-    #[allow(clippy::uninit_assumed_init)]
     fn default() -> Self {
-        let frames = unsafe { std::mem::MaybeUninit::uninit().assume_init() };
+        let frames = SmallVec::with_capacity(MAX_DEPTH);
         Self {
             frames,
-            depth: 0,
             thread_name: [0; MAX_THREAD_NAME],
             thread_name_length: 0,
             thread_id: 0,
         }
     }
 }
 
-impl Clone for UnresolvedFrames {
-    fn clone(&self) -> Self {
-        let slice = self.slice().clone();
-        Self::new(slice.frames, slice.thread_name, slice.thread_id)
-    }
-}
-
 impl Debug for UnresolvedFrames {
     fn fmt(&self, f: &mut Formatter) -> std::fmt::Result {
-        self.slice().fmt(f)
+        self.frames.fmt(f)
     }
 }
 
 impl UnresolvedFrames {
-    #[allow(clippy::uninit_assumed_init)]
-    pub fn new(bt: &[Frame], tn: &[u8], thread_id: u64) -> Self {
-        let depth = bt.len();
-        let mut frames: [Frame; MAX_DEPTH] =
-            unsafe { std::mem::MaybeUninit::uninit().assume_init() };
-        frames[0..depth].clone_from_slice(bt);
-
+    pub fn new(frames: SmallVec<[Frame; MAX_DEPTH]>, tn: &[u8], thread_id: u64) -> Self {
```

Comment:
Is it a breaking change?

Reply:
No. `UnresolvedFrames` is not public on crate level.